scripts/src/test/bash/integration-test-java-extra

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+cd ../test-setup
+CLI="${PWD}/scripts/devon"
+EXTRA_VERSION=$(head -n 1 "${PWD}"/mirrors/java/available-versions)
+
+echo "EXTRA_JAVA_VERSION=${EXTRA_VERSION}" >> "${PWD}/conf/devon.properties"
+"${CLI}" java setup
+
+if [ -x "${PWD}/software/extra/java/bin/java" ]
+then
+  rm "${PWD}/conf/devon.properties"
+  exit 0
+else
+  rm "${PWD}/conf/devon.properties"
+  exit 1
+fi
```

Comment:
same here.
If we always want to remove `devon.properties` to avoid side-effects between tests, shouldn't we just do this in all-tests?

For the record: `all-tests` is always creating a new folder for each test-case and creates a `conf/devon.properties` in there. It seems that all command-tests and all integration-tests always work in `test-setup`.
If that is our default, we could simply change this and create a central folder in `all-tests` where all tests are executed and that is properly initialized.
Then we could simplify all other tests that first `cd` to `test-setup`, deleting software (`rm -fR "${PWD}/software"`) that should be done in `all-tests`, etc.
Another thing is that most of these tests therefore "depend" on `test-setup` being run before.
However, this is not guaranteed for `test-*` as it depends on the directory order of the tests.

Reply:
I added this code to avoid redundancy with `cd test-setup` and `rm -fR software` and so on. In my view it is a stable change without changing the whole test structure. However, we can discuss whether to completely redesign the workflow of `all-tests`.

https://github.com/devonfw/ide/blob/bb49b7e547355351df8befe1ac151af36eea8e09/scripts/src/test/bash/all-tests#L14-L19

Reply:
Excellent solution. I am resolving this conversation.